homeassistant/components/husqvarna_automower/__init__.py

```diff
@@ -38,10 +48,21 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         await api_api.async_get_access_token()
     except ClientError as err:
         raise ConfigEntryNotReady from err
-    coordinator = AutomowerDataUpdateCoordinator(hass, automower_api)
+    coordinator = AutomowerDataUpdateCoordinator(hass, automower_api, entry)
     await coordinator.async_config_entry_first_refresh()
-
-    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = coordinator
+    init_ready = asyncio.Event()
+    listen_task = asyncio.create_task(
+        coordinator.client_listen(hass, entry, automower_api, init_ready)
+    )
+    try:
+        async with asyncio.timeout(LISTEN_READY_TIMEOUT):
+            await init_ready.wait()
+    except TimeoutError as err:
+        listen_task.cancel()
```

Comment:
We then don't need to cancel the task here. It will be done for us after we raise `ConfigEntryNotReady`.